worker/backend/backend_test.go

```diff
@@ -2,45 +2,57 @@ package backend_test
 
 import (
 	"errors"
+	"testing"
 
 	"github.com/concourse/concourse/worker/backend"
 	"github.com/concourse/concourse/worker/backend/libcontainerd/libcontainerdfakes"
-
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 )
 
-var _ = Describe("Backend", func() {
-
-	Describe("Ping", func() {
-
-		var (
-			err    error
-			be     backend.Backend
-			client *libcontainerdfakes.FakeClient
-		)
-
-		BeforeEach(func() {
-			client = new(libcontainerdfakes.FakeClient)
-			be = backend.New(client)
-		})
-
-		JustBeforeEach(func() {
-			err = be.Ping()
-		})
-
-		It("calls containerd's Version", func() {
-			Expect(client.VersionCallCount()).To(Equal(1))
+type BackendSuite struct {
+	suite.Suite
+	*require.Assertions
+
+	backend backend.Backend
+	client  *libcontainerdfakes.FakeClient
+}
+
+func (s *BackendSuite) SetupTest() {
+	s.client = new(libcontainerdfakes.FakeClient)
+	s.backend = backend.New(s.client)
+}
+
+func (s *BackendSuite) TestPing() {
+	for _, tc := range []struct {
+		versionReturn error
+		succeeds      bool
+	}{
+		{
+			versionReturn: nil,
+			succeeds:      true,
+		},
+		{
+			versionReturn: errors.New("errr"),
+			succeeds:      false,
+		},
+	} {
+		s.T().Run("case", func(t *testing.T) {
```

Comment:
this gets expanded to `case#1`, `case#2`, ...

Reply:
another pattern would be to take `desc` inside the "anonymous struct". wdyt?

Reply:
yeah the `desc` field pattern seems like a nice idea

Reply:
looks good indeed 😁  just applied